### ISSUE TITLE
[FEATURE] Automatic attach of custom metadata when uploading recordings to Amazon S3

### ISSUE THREAD

**Issue by JSON-OBJECT** [livekit/egress#518]
## TLDR
  * When uploading a recording file to Amazon S3 in LiveKit Egress, is it possible to add Content-Disposition: attachment as metadata?

## Detailed explanation and questions
  * In my project, I use Kotlin SDK to make a recording request to Amazon S3 through LiveKit Egress. In this case, I am using it as below code. Is it possible to automatically include Content-Disposition: attachment in the metadata when uploading all recorded files? The reason I need this is because I want the Save as feature when downloading from the Browser. If we want to add metadata after the upload is complete, must copy the S3 Object, which is inefficient.
  
```kotlin
        val output = LivekitEgress.EncodedFileOutput.newBuilder()
            .setFileType(LivekitEgress.EncodedFileType.MP4)
            .setFilepath(".../room-composite-{time}")
            .setS3(
                LivekitEgress.S3Upload.newBuilder()
                    .setAccessKey($AWS_ACCESS_KEY_ID)
                    .setSecret($AWS_SECRET_ACCESS_KEY)
                    .setRegion($AWS_REGION)
                    .setBucket($FFL_S3_BUCKET)
                    .build()
            )
            .build()
```
